src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,555 @@
+    ) public override {
+        _recoverAssets(from, to, gameId, assetIds);
+    }
+
+    /// @notice Check if the contract supports an interface.
+    /// 0x01ffc9a7 is ERC-165.
+    /// 0x80ac58cd is ERC-721.
+    /// @param id The id of the interface.
+    /// @return if the interface is supported.
+    function supportsInterface(bytes4 id) public pure override returns (bool) {
+        return id == 0x01ffc9a7 || id == 0x80ac58cd || id == 0x5b5e139f;
+    }
+
+    /// @notice Add assets to an existing GAME.
+    /// @param from The address of the current owner of assets.
+    /// @param baseId The baseId of the GAME to add assets to.
+    /// @param assetIds The id of the asset to add to GAME.
+    /// @param amounts The amount of each asset to add to GAME.
+    function _addAssets(
+        address from,
+        uint256 baseId,
+        uint256[] memory assetIds,
+        uint256[] memory amounts
+    ) internal {
+        if (assetIds.length == 0) {
+            return;
+        }
+        require(assetIds.length == amounts.length, "INVALID_INPUT_LENGTHS");
+        uint256 currentValue;
+        for (uint256 i = 0; i < assetIds.length; i++) {
+            currentValue = _gameAssets[baseId][assetIds[i]];
+            require(amounts[i] != 0, "INVALID_ASSET_ADDITION");
+            _gameAssets[baseId][assetIds[i]] = currentValue.add(amounts[i]);
+        }
+        if (assetIds.length == 1) {
+            _asset.safeTransferFrom(from, address(this), assetIds[0], amounts[0], "");
+        } else {
+            _asset.safeBatchTransferFrom(from, address(this), assetIds, amounts, "");
+        }
+    }
+
+    /// @notice Remove assets from a GAME.
+    /// @param baseId The baseId of the GAME to remove assets from.
+    /// @param assetIds An array of asset Ids to remove.
+    /// @param values An array of the number of each asset id to remove.
+    /// @param to The address to send removed assets to.
+    function _removeAssets(
+        uint256 baseId,
+        uint256[] memory assetIds,
+        uint256[] memory values,
+        address to
+    ) internal {
+        if (assetIds.length == 0) {
+            return;
+        }
+        require(assetIds.length == values.length && assetIds.length != 0, "INVALID_INPUT_LENGTHS");
+        uint256 currentValue;
+        for (uint256 i = 0; i < assetIds.length; i++) {
+            currentValue = _gameAssets[baseId][assetIds[i]];
+            require(currentValue != 0 && values[i] != 0 && values[i] <= currentValue, "INVALID_ASSET_REMOVAL");
+            _gameAssets[baseId][assetIds[i]] = currentValue.sub(values[i]);
+        }
+
+        if (assetIds.length == 1) {
+            _asset.safeTransferFrom(address(this), to, assetIds[0], values[0], "");
+        } else {
+            _asset.safeBatchTransferFrom(address(this), to, assetIds, values, "");
+        }
+    }
+
+    /// @dev See destroyGame.
+    function _destroyGame(
+        address from,
+        address to,
```

Comment:
we should remove the `to` argument as it is not used.
the public function should also remove it (except for the destroy and recover version)
I am also wonderring if we should call destroyGame `burnFrom` and make a `burn` version
I think we use that elsewhere

Reply:
Yeah, That pattern is used in erc721basetoken. I can make that happen... it's a pretty easy change.